### ISSUE TITLE
HACKING incorrect

### ISSUE THREAD

**Issue by elextr** [geany/geany#1689]
[This line](https://github.com/geany/geany/blob/4a5ccf748ac2ce13a6c71cc41b354c33b6abd715/HACKING#L743) is not right, `src/geany`  is a libtool script, not an executable, so it not possible to debug it.  And it should not be in `src` if the build is outside the tree.

**Comment by kugel-** [comment]
libttool wrapper can be debugged using "libtool e gdb --args src/geany"

out-of-tree builds are probably mentioned elsewhere, so that the rest of the document can assume in-tree build.

**Comment by elextr** [comment]
Ok, does that also modify the prefix somehow so Geany will find the configuration files in-tree, not in their install locations?

**Comment by b4n** [comment]
@elextr no, it's not really possible to do that without rebuilding the software (and actually knowing a lot about how it uses these paths)

**Comment by elextr** [comment]
Ok, so although libtool might make it run, it won't be much use since Geany won't find the config files.  Better to recommend debugging a properly installed version then?

**Comment by b4n** [comment]
I guess so; but for code-only changes it can be slightly easier to debug the not-yet-installed executable.  But we probably shouldn't recommend that indeed, as it's easier to overlook a change that needed installing.

**Comment by elextr** [comment]
Yeah, its not hard to install to a non-system $prefix for testing.

**Comment by elextr** [comment]
See also #1688 for more GDB related HACKING suggestions.

**Comment by kugel-** [comment]
@elextr is this still valid?

**Comment by elextr** [comment]
@kugel- doesn't look like it

**Comment by elextr** [comment]
Sorry answered the wrong question, doesn't look like its fixed, so is still valid
